kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -425,4 +425,32 @@ public void deactivateAllSubscriptionsForSubscriber(long callingNumber) {
         }
     }
 
+    @Override
+    public Subscriber getSubscriberByMother(final long motherId) {
+        SqlQueryExecution<Subscriber> queryExecution = new SqlQueryExecution<Subscriber>() {
+
+            @Override
+            public String getSqlQuery() {
+                return KilkariConstants.SELECT_SUBSCRIBERS_BY_MOTHER_ID_OID;
+            }
+
+            @Override
+            public Subscriber execute(Query query) {
+                query.setClass(Subscriber.class);
+                ForwardQueryResult fqr = (ForwardQueryResult) query.execute(motherId);
+
+                if (fqr.isEmpty()) {
+                    return null;
+                }
+
+                if (fqr.size() == 1) {
+                    return (Subscriber) fqr.get(0);
+                }
+
+                throw new IllegalStateException(String.format(KilkariConstants.MORE_THAN_ONE_SUBSCRIBER, motherId));
```

Comment:
this is a pretty dangerous design pattern. If your data ever ends up in this state in your db due to manual updates, this will blow up and never recover. You will fail the rest of the sync and potentially even more since the original call of this method is marked as transactional. 

I understand that you are following patterns established in this file by other people, but you should probably have a catch on this particular exception and log an alert for this particular record. I dont think you want to fail the whole sync because of this. 

Also, this will be compounded by exception retries (x3) by activemq.